### PR TITLE
Refactoring/121 add unit test that verifies version references

### LINKED ISSUE #121: Add unit test that verifies version references
## Acceptance Criteria

- Validates that rockspec file name contains correct version number
- Validates that rockspec internally references correct version number
- Validates other references like in integration tests

### PR DESCRIPTION
**Pull request by redcatbear**
Closes #121.

### CODE REVIEW

File: doc/changes/changes_1.3.0.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Exasol Row Level Security (Lua) 1.3.0, released 2022-08-??
+
+Code name:
+
+## Summary
+
+## Features
+
+* #121: Verify references
```
Comment (kaklakariada) [inline doc/changes/changes_1.3.0.md:9]:
```suggestion
* #121: Added tests to verify version references
```

File: spec/build_preconditions_spec.lua [hunk 2]
```diff
@@ -0,0 +1,46 @@
+package.path = "spec/?.lua;" .. package.path
+require("busted.runner")()
+local Reader = require("pom.Reader")
+
+local function get_project_base_path()
+    return debug.getinfo(1,"S").source:sub(2):gsub("[^/]*$", "") .. ".."
+end
+
+local reader = Reader:new(get_project_base_path() .. "/pom.xml")
```
Comment (kaklakariada) [inline spec/build_preconditions_spec.lua:9]:
```suggestion
local pom = Reader:new(get_project_base_path() .. "/pom.xml")
```
or `pom_reader`

File: spec/build_preconditions_spec.lua [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            assert(file, "Expected rockspec to have filename " .. rockspec_path .. " but file not found.")
+        end)
+
+        describe("version field", function()
+            it("is of type string", function()
+                assert.is.same("string", type(rockspec.version))
+            end)
+
+            it("starts with the same version number as the main artifact in the Maven POM file", function()
+                assert.matches(reader:get_artifact_version() .. "%-%d+", rockspec.version)
```
Comment (kaklakariada) [inline spec/build_preconditions_spec.lua:42]: 👍

File: spec/build_preconditions_spec.lua [hunk 4]
```diff
@@ -0,0 +1,46 @@
+package.path = "spec/?.lua;" .. package.path
+require("busted.runner")()
+local Reader = require("pom.Reader")
```
Comment (kaklakariada) [inline spec/build_preconditions_spec.lua:3]:
```suggestion
local PomReader = require("pom.Reader")
```
❓

File: spec/exasolrls/RlsAdapter_spec.lua [hunk 5]
```diff
@@ -5,7 +5,13 @@ local mockagne = require("mockagne")
 local adapter_capabilities = require("exasolrls.adapter_capabilities")
 local RlsAdapter = require("exasolrls.RlsAdapter")
 require("exasolrls.RlsAdapterProperties")
-local pom = require("spec.pom.reader")
+local Reader = require("spec.pom.Reader")
```
Comment (kaklakariada) [inline spec/exasolrls/RlsAdapter_spec.lua:8]:
```suggestion
local PomReader = require("spec.pom.Reader")
```
❓

File: spec/exasolrls/RlsAdapter_spec.lua [hunk 6]
```diff
@@ -5,7 +5,13 @@ local mockagne = require("mockagne")
...
+
+local function get_project_base_path()
+    return debug.getinfo(1,"S").source:sub(2):gsub("[^/]*$", "") .. "../.."
+end
+
+local reader = Reader:new(get_project_base_path() .. "/pom.xml")
```
Comment (kaklakariada) [inline spec/exasolrls/RlsAdapter_spec.lua:14]:
```suggestion
local pom = Reader:new(get_project_base_path() .. "/pom.xml")
```

File: spec/pom/Reader.lua [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    return instance
+end
+
+function Reader:_init(path)
+    local pom = assert(io.open(path, "r"), "Failed to open POM: " .. path)
+    repeat
+        local line = pom:read("*l")
+        self._artifact_version = string.match(line,"<version>%s*([0-9.]+)") or self._artifact_version
+        self._artifact_id = string.match(line,"<artifactId>%s*([-.%w]+)") or self._artifact_id
+    until (self._artifact_id and self._artifact_version) or (line == nil)
```
Comment (kaklakariada) [inline spec/pom/Reader.lua:26]: We should fail when we don't find version & artifact id.

File: spec/smoke_test_spec.lua [hunk 8]
```diff
@@ -1,10 +1,15 @@
 package.path = "target/?.lua;" .. package.path
 require("busted.runner")()
+local Reader = require("spec.pom.Reader")
 
-local pom = require("spec.pom.reader")
+local function get_project_base_path()
+    return debug.getinfo(1,"S").source:sub(2):gsub("[^/]*$", "") .. ".."
+end
```
Comment (kaklakariada) [inline spec/smoke_test_spec.lua:7]: This is duplicated three times.
Maybe add a method `read_default_pom` to the Reader?

File: doc/changes/changes_1.3.0.md [hunk 9]
```diff
@@ -0,0 +1,11 @@
+# Exasol Row Level Security (Lua) 1.3.0, released 2022-08-??
```
[the remark this answers is not in the document]
Comment (redcatbear) [reply doc/changes/changes_1.3.0.md:1]: Indeed. I need to get the "set properties" request working in RLS. Already works in EVSL.

File: doc/developer_guide/developer_guide.md [hunk 10]
```diff
@@ -38,7 +38,7 @@ sudo luarocks install remotelog
 sudo luarocks install luacov
 sudo luarocks install luacov-coveralls
 sudo luarocks install luacheck
-sudo luarocks install exaerror
+sudo luarocks install ExaError
```
[the remark this answers is not in the document]
Comment (redcatbear) [reply doc/developer_guide/developer_guide.md:41]: Global search-and-replace is dangerous. Thanks for the hint. Fixed.

File: spec/exasolrls/RlsAdapter_spec.lua (hunk 6 above)
Comment (redcatbear) [reply spec/exasolrls/RlsAdapter_spec.lua:14]: Changed to `pom`.

File: spec/exasolrls/RlsAdapter_spec.lua (hunk 5 above)
Comment (redcatbear) [reply spec/exasolrls/RlsAdapter_spec.lua:8]: Changed from `pom.Reader` to `PomReader`

File: spec/pom/Reader.lua (hunk 7 above)
Comment (redcatbear) [reply spec/pom/Reader.lua:26]: Added.

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**